typescript/src/deposit-refund.ts

```diff
@@ -0,0 +1,336 @@
+import bcoin from "bcoin"
+import { BigNumber } from "ethers"
+import {
+  createKeyRing,
+  RawTransaction,
+  Client as BitcoinClient,
+  TransactionHash,
+  UnspentTransactionOutput,
+  computeHash160,
+  isCompressedPublicKey,
+} from "./bitcoin"
+import {
+  assembleDepositScript,
+  Deposit,
+  validateDepositScriptParameters,
+} from "./deposit"
+
+/**
+ * Submits a deposit refund by creating and broadcasting a Bitcoin P2(W)PKH
+ * deposit refund transaction.
+ * @param bitcoinClient - Bitcoin client used to interact with the network.
+ * @param fee - the value that will be subtracted from the deposit UTXO being
+ *        refunded and used as the transaction fee.
+ * @param utxo - UTXO that was created during depositing that needs be refunded.
+ * @param deposit - Details of the deposit being refunded. It should contain
+ *        the same data that was used during depositing.
+ * @param refunderAddress - Address the recipient of the refunded deposit.
+ * @param refunderPrivateKey - Bitcoin private key of the refunder. It must
+ *        correspond to the `refundPublicKeyHash` of the deposit script.
+ * @returns The outcome is the deposit refund transaction hash.
+ * @dev This function should be called by the refunder after `refundLocktime`
+ *      plus about 1 hour. The additional hour of waiting is the result of
+ *      adopting BIP113 which compares the transaction's locktime against the
+ *      median timestamp of the last 11 blocks. This median time lags
+ *      the current unix time by about 1 hour.
+ */
+export async function submitDepositRefundTransaction(
+  bitcoinClient: BitcoinClient,
+  fee: BigNumber,
+  utxo: UnspentTransactionOutput,
+  deposit: Deposit,
+  refunderAddress: string,
+  refunderPrivateKey: string
+): Promise<{ transactionHash: TransactionHash }> {
+  const utxoRawTransaction = await bitcoinClient.getRawTransaction(
+    utxo.transactionHash
+  )
+
+  const utxoWithRaw = {
+    ...utxo,
+    transactionHex: utxoRawTransaction.transactionHex,
+  }
+
+  const { transactionHash, rawTransaction } =
+    await assembleDepositRefundTransaction(
+      fee,
+      utxoWithRaw,
+      deposit,
+      refunderAddress,
+      refunderPrivateKey
+    )
+
+  // Note that `broadcast` may fail silently (i.e. no error will be returned,
+  // even if the transaction is rejected by other nodes and does not enter the
+  // mempool, for example due to an UTXO being already spent).
+  await bitcoinClient.broadcast(rawTransaction)
```

Comment:
Can we wrap this call in the `try catch`? Or are you saying that even `try catch` won't catch an error?

Reply:
There will be no errors. The function will fail silently.